src/main/java/bio/terra/service/snapshot/SnapshotDao.java

```diff
@@ -558,17 +486,17 @@ class RawSourceData {
       SnapshotSource snapshotSource =
           new SnapshotSource().id(raw.id).snapshot(snapshot).dataset(dataset);
 
-      if (raw.assetId != null) { // if there is no assetId, then dont check for a spec
+      if (raw.assetId != null) { // if there is no assetId, then don't check for a spec
         // Find the matching asset in the dataset
         Optional<AssetSpecification> assetSpecification =
             dataset.getAssetSpecificationById(raw.assetId);
-        if (!assetSpecification.isPresent()) {
+        if (assetSpecification.isEmpty()) {
           throw new CorruptMetadataException("Asset referenced by snapshot source was not found!");
         }
         snapshotSource.assetSpecification(assetSpecification.get());
```

Comment:
I think it's best to avoid calling `Optional.get` whenever possible, since doing that means you're separating the `isPresent()` from the `get()`. In this case you can use `orElseThrow()`
```suggestion
        var assetSpecification = dataset.getAssetSpecificationById(raw.assetId);
        snapshotSource.assetSpecification(assetSpecification.orElseThrow(() ->
            new CorruptMetadataException("Asset referenced by snapshot source was not found!")));
```

Or, if you want to go even further toward `Optional`, this should work too (back up to and including the `if (raw.assetId ...)`)
```
      Optional.ofNullable(raw.assetId)
          .flatMap(dataset::getAssetSpecificationById)
          .map(snapshotSource::assetSpecification)
          .orElseThrow(() -> new CorruptMetadataException("Asset referenced by snapshot source was not found!"));
```

Reply:
I like this much better, thanks!